robot-core/src/main/java/org/obolibrary/robot/ExpandOperation.java

```diff
@@ -0,0 +1,150 @@
+
+  public static class ExpandConfig {
+
+    private boolean createNew = false;
+    private boolean annotateExpansions = false;
+    private final Set<IRI> expandProperties = new HashSet<>();
+    private final Set<IRI> excludeProperties = new HashSet<>();
+
+    public boolean createNewOntology() {
+      return createNew;
+    }
+
+    public void setCreateNewOntology(boolean shouldCreateNew) {
+      this.createNew = shouldCreateNew;
+    }
+
+    public boolean annotateExpansionAxioms() {
+      return annotateExpansions;
+    }
+
+    public void setAnnotateExpansionAxioms(boolean shouldAnnotate) {
+      this.annotateExpansions = shouldAnnotate;
+    }
+
+    public Set<IRI> getExpandProperties() {
+      return Collections.unmodifiableSet(expandProperties);
+    }
+
+    public void setExpandProperties(Set<IRI> properties) {
+      expandProperties.clear();
+      expandProperties.addAll(properties);
+    }
+
+    public Set<IRI> getExcludeProperties() {
+      return Collections.unmodifiableSet(excludeProperties);
+    }
+
+    public void setExcludeProperties(Set<IRI> properties) {
+      excludeProperties.clear();
+      excludeProperties.addAll(properties);
+    }
+  }
+
+  /**
+   * Expand macro relations using SPARQL CONSTRUCT
+   *
+   * @param ontology the ontology to expand from
+   * @param config configuration for expansion operation
+   * @param includeTerms IRIs of annotation properties whose expansion to perform. If empty, all
+   *     found expansion are included.
+   * @param excludeTerms IRIs of annotation properties whose expansion to exlude; subtracted from
+   *     inclusion list.
+   * @throws OWLOntologyStorageException
+   * @throws IOException
+   */
+  public static void expand(
+      OWLOntology ontology, ExpandConfig config, Set<IRI> includeTerms, Set<IRI> excludeTerms)
+      throws OWLOntologyStorageException, IOException {
+    Dataset dataset = QueryOperation.loadOntologyAsDataset(ontology, true);
+    Set<OWLAxiom> expansions =
+        ontology.getAxioms(AxiomType.ANNOTATION_ASSERTION, Imports.INCLUDED).stream()
+            .filter(ax -> ax.getProperty().equals(definedByConstruct))
+            .filter(ax -> ax.getSubject().isIRI())
+            .filter(ax -> includeTerms.isEmpty() || includeTerms.contains(ax.getSubject()))
+            .filter(ax -> !excludeTerms.contains(ax.getSubject()))
+            .filter(ax -> ax.getValue().isLiteral())
+            .map(
+                ax ->
+                    executeConstruct(
+                        dataset,
+                        ax.getValue().asLiteral().orNull().getLiteral(),
+                        (IRI) (ax.getSubject()),
+                        config.annotateExpansionAxioms()))
+            .reduce(ExpandOperation::combine)
+            .orElse(Collections.emptySet());
+    if (config.createNew) {
+      ontology.getOWLOntologyManager().removeAxioms(ontology, ontology.getAxioms());
```

Comment:
Lol, I didnt know this is how we do this.. I would have thought an entirely new ontology, but, why not. I can see this helps with piping. Sorry ignore comment just noise

Reply:
I followed the `reason` implementation for that.